idseq_dag/util/server.py

```diff
@@ -176,3 +180,21 @@ def wait_for_server_ip(service_name,
                                          max_concurrent, chunk_id)
         return result
 
+
+def get_instance_iD_from_iP(instance_iP):
```

Comment:
Don't do this on every chunk -- it's slow and we will likely run into rate limit issues as the number of concurrent jobs and chunks goes into the hundreds.    I recommend instead you modify get_server_ips_work so it remembers and returns this information along with the server ips.